File: src/problems_all/191_number_of_1_bits/191_number_of_1_bits_v2.py
"""
https://leetcode.com/problems/number-of-1-bits/description/

Runtime: 31 ms
Memory: 13.9 MB
"""


class Solution:
    def hammingWeight(self, n: int) -> int:
        _bin = format(n, 'b')
        digits = list(_bin)
        return digits.count('1')


def main():
    n = 123
    weight = Solution().hammingWeight(n)
    print(weight)


if __name__ == '__main__':
    main()

File: src/problems_all/191_number_of_1_bits/191_number_of_1_bits_v2.ts
/*
https://leetcode.com/problems/number-of-1-bits/description/

Runtime: 74 ms
Memory: 44.2 MB
https://leetcode.com/problems/number-of-1-bits/submissions/874795349/
*/


function hammingWeight(n: number): number {
  let bin = (n >>> 0).toString(2);
  let count = 0;
  for (let c of bin) {
    if (c === '1') {
      count++;
    }
  }

  return count;
}

function main(): void {
  let n = 2147483647;
  let result = hammingWeight(n);
  console.log(result);
}

main();
